src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -78,20 +89,33 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	// Establish outbound connection.
-	addr := hostIncludingPort(req, 443)
-	d := client.proxiedDialer(func(network, addr string) (net.Conn, error) {
-		// UGLY HACK ALERT! In this case, we know we need to send a CONNECT request
-		// to the chained server. We need to send that request from chained/dialer.go
-		// though because only it knows about the authentication token to use.
-		// We signal it to send the CONNECT here using the network transport argument
-		// that is effectively always "tcp" in the end, but we look for this
-		// special "transport" in the dialer and send a CONNECT request in that
-		// case.
-		return client.getBalancer().Dial("connect", addr)
-	})
-
-	connOut, err = d("tcp", addr)
+	sendToProxy := false
+	for _, proxiedPort := range client.ProxiedCONNECTPorts {
+		if port == proxiedPort {
+			sendToProxy = true
+			break
+		}
+	}
```

Comment:
I agree that we need better coverage, and that's a good idea.  Do you mind if I handle this as part of #3685 so I can approach it holistically?